src/components/exchange/exchangeAssetRowContextMenuProps.js

```diff
@@ -0,0 +1,102 @@
+import lang from 'i18n-js';
+import { startCase } from 'lodash';
+import { setClipboard } from '../../hooks/useClipboard';
+import {
+  abbreviations,
+  ethereumUtils,
+  haptics,
+  showActionSheetWithOptions,
+} from '@rainbow-me/utils';
+
+const buildBlockExplorerAction = type => {
+  const blockExplorerText = lang.t('exchange.coin_row.view_on', {
+    blockExplorerName: startCase(ethereumUtils.getBlockExplorer(type)),
+  });
+  return {
+    actionKey: CoinRowActionsEnum.blockExplorer,
+    actionTitle: blockExplorerText,
+    icon: {
+      iconType: 'SYSTEM',
+      iconValue: 'link',
+    },
+  };
+};
+
+const CoinRowActionsEnum = {
+  blockExplorer: 'blockExplorer',
+  copyAddress: 'copyAddress',
+};
+
+const CoinRowActions = {
+  [CoinRowActionsEnum.copyAddress]: {
+    actionKey: CoinRowActionsEnum.copyAddress,
+    actionTitle: lang.t('wallet.action.copy_contract_address'),
+    icon: {
+      iconType: 'SYSTEM',
+      iconValue: 'doc.on.doc',
+    },
+  },
+};
+
+export default function contextMenuProps(item, onCopySwapDetailsText) {
```

Comment:
I don't think we should introduce more JS files in our codebase so could you please migrate this file to typescript

Reply:
fixed